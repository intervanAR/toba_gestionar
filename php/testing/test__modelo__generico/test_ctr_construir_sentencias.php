<?php

/**
 * @author lgraziani
 */
class test_ctr_construir_sentencias extends toba_test
{
	private static $snap;

	public function setUp()
	{
		self::$snap = new snapshot_tester(__DIR__, $this);
	}

	/**
	 * @override
	 */
	public static function get_descripcion()
	{
		return 'modelo/generico/ctr_construir_sentencias';
	}

	public function test__sql_update()
	{
		$tabla = 'RE_CALLES';
		$columnas = [
			'columna1' => 'valor1',
			'columna2' => 2,
			'columna3' => 'valor3',
			'columna4' => 4,
		];
		$where = 'id = 1';
		$sql = ctr_construir_sentencias::sql_update($tabla, $columnas, $where);

		self::$snap->match_snapshot($sql);
	}

	public function test__construir_sentencia_insert()
	{
		$tabla = 'RE_CALLES';
		$columnas = [
			'columna1' => 'valor1',
			'columna2' => 2,
			'columna3' => 'valor3',
			'columna4' => 4,
		];
		$sql = ctr_construir_sentencias::construir_sentencia_insert($tabla, $columnas);

		self::$snap->match_snapshot($sql);
	}
}
