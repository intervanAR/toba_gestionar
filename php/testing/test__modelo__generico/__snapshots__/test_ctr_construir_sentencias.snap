a:2:{s:16:"test__sql_update";s:106:"UPDATE RE_CALLES SET columna1 = 'valor1', columna2 = '2', columna3 = 'valor3', columna4 = '4' WHERE id = 1";s:32:"test__construir_sentencia_insert";s:106:"INSERT INTO RE_CALLES (columna1 , columna2 , columna3 , columna4) VALUES ('valor1' , '2' , 'valor3' , '4')";}